anchor/src/instruction.rs

```diff
@@ -0,0 +1,102 @@
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+
+use solana_program::{
+    account_info::AccountInfo,
+    instruction::{AccountMeta, Instruction},
+    program_error::ProgramError,
+    pubkey::Pubkey,
+    sysvar,
+};
+
+use lido::{accounts_struct, accounts_struct_meta, error::LidoError, token::StLamports};
+
+#[repr(C)]
+#[derive(Clone, Debug, BorshSerialize, BorshDeserialize, BorshSchema)]
+pub enum AnchorInstruction {
+    Initialize,
+
+    /// Deposit a given amount of StSOL, gets bSOL in return.
+    ///
+    /// This can be called by anybody.
+    Deposit {
+        #[allow(dead_code)] // but it's not
+        amount: StLamports,
+    },
+
+    /// Withdraw a given amount of stSOL.
+    ///
+    /// Caller provides some `amount` of StLamports that are to be burned in
+    /// order to withdraw bSOL.
+    Withdraw {
+        #[allow(dead_code)] // but it's not
+        amount: StLamports,
+    },
+
+    /// Claim rewards on Terra.
+    ClaimRewards,
+}
+
+impl AnchorInstruction {
+    pub fn to_vec(&self) -> Vec<u8> {
+        // `BorshSerialize::try_to_vec` returns a Result, because it uses
+        // `Borsh::serialize`, which takes an arbitrary writer, and which can
+        // therefore return an IoError. But when serializing to a vec, there
+        // is no IO, so for this particular writer, it should never fail.
+        self.try_to_vec()
+            .expect("Serializing an Instruction to Vec<u8> does not fail.")
+    }
+}
+
+accounts_struct! {
+    InitializeAccountsMeta, InitializeAccountsInfo {
+        pub anchor {
```

Comment:
For the Lido program we were too late, but for the Anchor integration program, we can use this trick that Neodyme pointed out: make the instance address a program-derived address. We can use the Lido instance address as seed for the derivation. That way every Lido instance will have one unique associated Anchor instance.

(Nothing needs to be changed on this line for that, I’m just writing down the idea.)